src/lib/routes/admin-api/user-splash-controller.ts

```diff
@@ -0,0 +1,78 @@
+import { Response } from 'express';
+
+import Controller from '../controller';
+import { Logger } from '../../logger';
+import { IUnleashConfig } from '../../types/option';
+import { IUnleashServices } from '../../types/services';
+import UserSplashService from '../../services/user-splash-service';
+import { IAuthRequest } from '../unleash-types';
+
+interface ISplashBody {
+    seen: boolean;
+    splashId: string;
+}
+
+class UserSplashController extends Controller {
+    private logger: Logger;
+
+    private userSplashService: UserSplashService;
+
+    constructor(
+        config: IUnleashConfig,
+        { userSplashService }: Pick<IUnleashServices, 'userSplashService'>,
+    ) {
+        super(config);
+        this.logger = config.getLogger('splash-controller.ts');
+        this.userSplashService = userSplashService;
+
+        this.post('/', this.recordSplash);
+        this.put('/:id', this.updateSplashSettings);
+    }
+
+    private async recordSplash(
```

Comment:
This does the same as the /:id endpoint. I think the id endpoint makes more sense in this context.